plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceRuntimeInfo.java

```diff
@@ -208,16 +205,13 @@ public String projectsRoot() {
 
     @Override
     public Map<String, ServerImpl> getServers() {
-        try {
-            return addDefaultReferenceForServersWithoutReference(
-                    addRefAndUrlToServers(getServersWithFilledPorts(containerHost,
-                                                                    info.getNetworkSettings().getPorts()),
-                                          info.getConfig().getLabels()));
-        } catch (Exception e) {
-            // Hack needed to investigate why swarm returns invalid response
-            LOG.error(info.toString());
-            throw e;
-        }
+        Map<String, List<PortBinding>> ports = info.getNetworkSettings().getPorts();
+        Map<String, String> labels = info.getConfig().getLabels();
```

Comment:
Looks like they are never null whereas ports field is null when container is not running